dm/worker/config.go

```diff
@@ -256,8 +261,52 @@ func (c *Config) configFromFile(path string) error {
 	return c.verify()
 }
 
+func (c *Config) adjust() error {
+	c.From.Adjust()
+	c.Checker.adjust()
+
+	if c.Flavor == "" || c.ServerID == 0 {
+		fromDB, err := c.createBaseDB()
+		if err != nil {
+			return err
+		}
+		defer fromDB.Close()
+
+		ctx, cancel := context.WithTimeout(context.Background(), dbGetTimeout)
+		defer cancel()
+
+		err = c.adjustFlavor(ctx, fromDB.DB)
+		if err != nil {
+			return err
+		}
+
+		err = c.adjustServerID(ctx, fromDB.DB)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *Config) createBaseDB() (*conn.BaseDB, error) {
```

Comment:
use a meaningful name, like `createFromDB`? maybe we need a `toDB` someday.

Reply:
update 354a2fc